sdk/identity/Azure.Identity/src/Credentials/BrowserCustomizationOptions.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+using Microsoft.Identity.Client;
+
+namespace Azure.Identity
+{
+    /// <summary>
+    /// Optiones to customize browser view.
+    /// </summary>
+    /// <seealso cref="BrowserCustomizationOptions"/>.
+    public class BrowserCustomizationOptions
+    {
+        internal bool? UseEmbeddedWebView;
+
+        internal SystemWebViewOptions SystemBrowserOptions;
+
+        internal EmbeddedWebViewOptions EmbeddedBrowserOptions;
+
+        private SystemWebViewOptions systemWebViewOptions
+        {
+            get
+            {
+                if (SystemBrowserOptions == null)
+                {
+                    SystemBrowserOptions = new SystemWebViewOptions();
+                }
+                return SystemBrowserOptions;
+            }
+        }
+
+        private EmbeddedWebViewOptions embeddedWebViewOptions
+        {
+            get
+            {
+                if (EmbeddedBrowserOptions == null)
+                {
+                    EmbeddedBrowserOptions = new EmbeddedWebViewOptions();
+                }
+                return EmbeddedBrowserOptions;
+            }
+        }
+
+        /// <summary>
+        /// Default constructor for BrowserCustomizationOptions. Will keep the framework default behaviour,
+        /// when use this constructor.
+        /// </summary>
+        public BrowserCustomizationOptions()
+        {
+            UseEmbeddedWebView = null;
+            SystemBrowserOptions = null;
+            EmbeddedBrowserOptions = null;
+        }
+
+        /// <summary>
+        /// The constructor to customize system OS browser.
+        /// </summary>
+        public BrowserCustomizationOptions(
+            Uri browserRedirectError,
+            Uri browserRedirectSuccess,
+            string htmlMessageSuccess,
+            string htmlMessageError,
+            bool iOSHidePrivacyPrompt,
+            Func<Uri, System.Threading.Tasks.Task> openBrowserAsync):this()
```

Comment:
> Do we need all of these options? I was thinking of simplifying this to a single bool added to `InteractiveBrowserCredentialOptions` called `UseEmbeddedWebView`

According to current requirement from PM, we need `htmlMessageSuccess` and `browserRedirectSuccess`. But shall we add all the options in case the requirements change in the future?

Reply:
@Alex-wdy Can you please confirm the requirements?

Reply:
> @Alex-wdy Can you please confirm the requirements?

Hi @scottaddie @christothes, we now implement the feature by setting html docs to `HtmlMessageSuccess` and `HtmlMessageError`. I have revised PR accordingly. Please take a look again, thanks.